src/bgg/sampler.rs

```diff
@@ -8,14 +8,15 @@ use crate::{
     utils::debug_mem,
 };
 use rayon::prelude::*;
-use std::{marker::PhantomData, sync::Arc};
+use std::marker::PhantomData;
 
 /// A sampler of a public key A in the BGG+ RLWE encoding scheme
 #[derive(Clone)]
 pub struct BGGPublicKeySampler<K: AsRef<[u8]>, S: PolyHashSampler<K>> {
-    pub sampler: Arc<S>,
```

Comment:
I think it is fine to keep the `sampler` in the struct.
This can be automatically made in the new function

Reply:
I think its cleaner to remove sampler to not store under struct

Reply:
Note for future updates: if a different sampler implementation built in the future (e.g., GPU based implementation) contains some heavy precomputation results within the struct, we should keep the sampler struct in the `BGGPublicKeySampler`.